lib/rubocop/cop/style/endless_method_operator_precedence.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Looks for endless methods inside operations of lower precedence (`and`, `or`, and
+      # modifier forms of `if`, `unless`, `while`, `until`) that are ambiguous due to
+      # lack of parentheses. This may lead to unexpected behavior as the code may appear
+      # to use these keywords as part of the method but in fact they modify
+      # the method definition itself.
+      #
+      # @safety
+      #   Autocorrection is unsafe as it changes potentially intended behavior; ie. if it
+      #   is actually desired to have the method definition modified by one of the keywords.
+      #   In this case, wrapping the method definition in parens will avoid the cop.
+      #
+      # @example
+      #
+      #   # bad
+      #   def foo = true and false
+      #
+      #   # good
+      #   def foo = (true and false)
+      #
+      #   # bad
+      #   def foo = true if bar
+      #
+      #   # good
+      #   def foo = (true if bar)
+      class EndlessMethodOperatorPrecedence < Base
+        extend TargetRubyVersion
+        extend AutoCorrector
+        include RangeHelp
+
+        minimum_target_ruby_version 3.0
+
+        MSG = 'Use parentheses around `%<keyword>s` statement in endless method body.'
+
+        # @!method ambiguous_endless_method_body(node)
+        def_node_matcher :ambiguous_endless_method_body, <<~PATTERN
+          ^${
+            (if _ <def _>)
```

Comment:
What about `unless` ?

Reply:
There is no specific `unless` node, the unless modifier parses into an `if` node with only an "else" clause.

```
$ ruby-parse --legacy -e 'def foo = true unless bar'
(if
  (send nil :bar) nil
  (def :foo
    (args)
    (true)))
```